crypto/evp/mac_lib.c

```diff
@@ -112,8 +112,6 @@ int EVP_MAC_init(EVP_MAC_CTX *ctx)
 
 int EVP_MAC_update(EVP_MAC_CTX *ctx, const unsigned char *data, size_t datalen)
 {
-    if (datalen == 0)
```

Comment:
Did this break something?

Reply:
Yes. We assume that when doing TLS-HMAC we call EVP_MAC_update exactly twice. The first time contains the TLS header data, and the second time the record data itself. The `datalen` is the size of the data after padding and mac removal, but "data-size" is the length including padding and mac. Unfortunately in the case where datalen is 0, it doesn't get passed down to the underlying implementation (even though data-size is non-zero). This particularly happens in TLS 1.0 where we don't have explicit IVs in records. The IV is implicit and this can lead to attacks. As a mitigation every other record we send has zero payload length.